SunbirdAI/noise-sensors-monitoring
# Add device metrics feature

This PR adds **device metrics** features to the Noise Monitoring Dashbaord

**Completed tasks:**
* Listing each device's metrics on the UI
* Endpoints for:
   * Creating a device metric
   * Retrieving all metrics (from all devices)
   * Retrieving metrics from a particular device

---

**How to test this PR:**
- Pull this branch
- Run the app: `docker-compose -f docker-compose-web.yml up`
- To test **creating device metrics using the API endpoint**: in Postman, run `POST http://localhost:8000/device_metrics/`. The request body should be structured as shown in the example below:
```
 {
     "device": "SB1001",
     "sig_strength": 7,
     "db_level": 5,
     "last_rec": 4,
     "last_upl": 4,
     "panel_voltage": 4,
     "battery_voltage": 4,
     "data_balance": 4
 }
```
- To test **listing all device metrics (for all devices) using the API endpoint**: in Postman, run `GET http://localhost:8000/device_metrics/` in Postman or your browser
- To test **retrieving device metrics for a particular device (filtered by device_id) using the API endpoint**: in Postman, run `http://localhost:8000/device_metrics/<device_id>`. 
For example: `POST http://localhost:8000/device_metrics/SB1001`
- To **view a device's metrics on the UI**, go to `http://localhost:8000/devices/<uuid>`
